### ISSUE TITLE
Fine-tuning the softmax layer with custom relations

### ISSUE THREAD

**Issue by Iqra840** [Babelscape/rebel#42]
Hello :) Thank  you first of all for all your work on the relation extraction domain.

I would like to ask if it is possible to add a softmax layer to the REBEL model architecture that can assign a probability to a list of candidate relations about which is the most fitting. The purpose of doing this is to have only a few dozen custom relations to choose from in the final stage- do you think this would be possible?
Alternatively, is there any way you would recommended filtering the relation list of the pretrained REBEL model to have a user-defined list of about 100 relations? Any guidance regarding this would be very helpful.

Thanks again!

**Comment by LittlePea13** [comment]
Hi there! Thanks for using REBEL :)

At each decoding step, you are selecting from any possible word in your vocabulary (already after a Softmax layer, so you already have probabilities). In my experience, there is no need to constrain anything, but if you want to get probabilities for specific relations, you could compute the perplexity for a set of relations you are interested on. However, take into account that this would be conditioned on all the triplets already decoded, and you would have to compute it each time a subject and object are decoded.

So if you have a set of 100 relations within the REBEL dataset (or a dataset of your choice with enough training data) I suggest you finetune for a bit, depending on whether those relations are already in the REBEL pretraining or not, and the model should then only predict relations within those seen at finetuning time, but would take advantage of the pretraining phase, as we did with datasets such as NYT or CONLL04.

Best of luck,
Pere-Lluis.
